homeassistant/components/home_connect/light.py

```diff
@@ -104,7 +104,7 @@ async def async_turn_on(self, **kwargs):
                     if ATTR_BRIGHTNESS in kwargs:
                         brightness = 10 + ceil(kwargs[ATTR_BRIGHTNESS] / 255 * 90)
 
-                    hs_color = kwargs.get(ATTR_HS_COLOR, default=self._hs_color)
+                    hs_color = kwargs[ATTR_HS_COLOR]
```

Comment:
is `hs_color` is not provided, this will blow up now.

Reply:
A value will be assigned to hs_color so what do you mean with: is `hs_color` is not provided?

kwargs.get will give an exception: TypeError: get() takes no keyword arguments.

Not sure how to solve this otherwise.

Reply:
The turn on service can be called without setting a color, thus the keyword argument will be missing in that case. Calling it with `kwargs[ATTR_HS_COLOR]` will throw an error in that case as well.

I think this might be what you are looking for?

```py
hs_color = kwargs.get(ATTR_HS_COLOR, self._hs_color)
```

Reply:
Thanks for the clarification.
`hs_color = kwargs.get(ATTR_HS_COLOR, self._hs_color)` is the main reason for this PR. 
`hs_color = kwargs.get(ATTR_HS_COLOR, self._hs_color)` throughs an exception please refer to #45015.

Line 92 checks for type ambient lighting which contains both brightness and color.
Line 93 checks if `ATTR_HS_COLOR` is part of the turn on service,  If not then it skips `hs_color = kwargs[ATTR_HS_COLOR]` and executes simple a turn on in line 125.

Reply:
> Thanks for the clarification.
> `hs_color = kwargs.get(ATTR_HS_COLOR, self._hs_color)` is the main reason for this PR.
> `hs_color = kwargs.get(ATTR_HS_COLOR, self._hs_color)` throughs an exception please refer to #45015.

That is an incorrect statement.
`hs_color = kwargs.get(ATTR_HS_COLOR, default=self._hs_color)` is a reason for this PR.
Which is not what I suggested and not what you quoted.

> 
> Line 92 checks for type ambient lighting which contains both brightness and color.
> Line 93 checks if `ATTR_HS_COLOR` is part of the kwargs, If not then it skips `hs_color = kwargs[ATTR_HS_COLOR]` and executes a simple turn on in line 125.
> 
> The change to `kwargs[ATTR_HS_COLOR]` fixed issue #45015 (see comments).


That is also an incorrect statement.

Line 93: `if ATTR_BRIGHTNESS in kwargs or ATTR_HS_COLOR in kwargs:`

Is an `or` statement. This is also evaluated true if only brightness is provided. Hence your current suggested change will cause issues in those cases.

Reply:
Many sorries I misread the `default =` part. 
Thanks for correcting me.
I have pushed the change to dev.